### ISSUE TITLE
Server console does not show clients question callback or output of executed commands, servers execute commands meant for clientside/to effect client as if for the server/client hosted server

### ISSUE THREAD

**Issue by NilanthAnimosus** [Regalis11/Barotrauma#482]
Version 0.8.1.4

Currently servers do not respond to clients executing commands with the output that would be shown on the server side. Additionally different clients can answer other clients executed commands questions (Such as a banip command asking for a IP or duration, and another client types heal or something).

Additionally a dozen commands do not have code to make them work better with clients executing them, an example of this is "spawnitem X inventory" will spawn into the inventory of the host character instead of the client executing the command.

very often when spawning to cursor (Or any other cursor based command) their is a strong offset from the mouse position likely due to time since command vs velocity on the server side and latency often resulting in different rooms entirely in distance, so it would be useful for aspects such as "Inventory" to work.

Clients currently have "Executing server side command" or something if I recall, this isn't as helpful as the server responding with the servers output from that command, nor helpful the server does not yet keep track of whose question likely belongs to which client and not interfering with the hosts actual console.

**Comment by Regalis11** [comment]
I may have missed something but this only seems to be a problem in a few specific commands (banip, banid, kickid), the rest handle the question callbacks correctly (only showing them client-side and sending the end result to the server). I could not reproduce the issue with the client commands interfering with the host's console or other players' question callbacks - the callback is only invoked when the command is executed with the ExecuteCommand method, which is not used when a command is executed by a client.

**Comment by NilanthAnimosus** [comment]
I hadn't the time to mass-test every command, more what was noticed by hosts and what I figured may have caused issues elsewhere. I had tested the giveitem command, and figured it may effect others but hadn't tested them. I figured the callbacks executing on the server-side would be disruptive (For banip/banid/kickid) if they really were questions on the servers console instead of the clients.
